core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -920,8 +908,23 @@ protected SearchService newSearchService(ClusterService clusterService, IndicesS
         return customNameResolvers;
     }
 
-    /** Constructs an internal node used as a client into a cluster fronted by this tribe node. */
-    protected Node newTribeClientNode(Settings settings, Collection<Class<? extends Plugin>> classpathPlugins, Path configPath) {
+    public static class NodeBuilder {
```

Comment:
I don't think we need this entire construct we can just subclass Node.java in the plugin / module and create a node via this?